app/src/main/java/com/example/keniphhomemaintenance/maintenance_screen/MaintenanceItemDialog.kt

```diff
@@ -0,0 +1,122 @@
+    maintenanceViewModel: MaintenanceViewModel = viewModel()
+) {
+    val itemTitle = remember { mutableStateOf("") }
+    val dueDate = remember { mutableStateOf("") }
+    val location = remember { mutableStateOf("") }
+
+    Dialog(
+        onDismissRequest = { isDialogOpen.value = false }) {
+        Surface(
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(vertical = MAINTENANCE_ITEM_DIALOG_VERTICAL_PADDING.dp),
+            shape = RoundedCornerShape(MAINTENANCE_ITEM_DIALOG_CORNER_RADIUS.dp),
+            color = MaterialTheme.colors.background
+        ) {
+            Column(
+                modifier = Modifier.padding(MAINTENANCE_ITEM_DIALOG_TOP_PADDING.dp),
+                horizontalAlignment = Alignment.CenterHorizontally,
+                verticalArrangement = Arrangement.Top
+            ) {
+                Text(
+                    text = stringResource(R.string.maintenance_item_dialog_title),
+                    color = MaterialTheme.colors.onBackground,
+                    fontWeight = FontWeight.Bold,
+                    fontSize = MAINTENANCE_ITEM_CARD_TITLE_TEXT_SIZE.sp
+                )
+                Spacer(modifier = Modifier.padding(MAINTENANCE_ITEM_DIALOG_TITLE_BOTTOM_PADDING.dp))
+
+                MaintenanceTextField(
+                    fieldValue = itemTitle,
+                    fieldLabel = R.string.task_name_field_label,
+                    placeholder = R.string.maintenance_item_dialog_task_name_field_placeholder,
+                    focusAction = FocusOptions.Down
+                )
+
+                Spacer(modifier = Modifier.padding(MAINTENANCE_ITEM_DIALOG_PADDING_BETWEEN_TEXT_FIELDS.dp))
+
+                MaintenanceTextField(
+                    fieldValue = dueDate,
+                    fieldLabel = R.string.due_date_field_label,
+                    placeholder = R.string.maintenance_item_dialog_due_date_field_placeholder,
+                    focusAction = FocusOptions.Down
+                )
+
+                Spacer(modifier = Modifier.padding(MAINTENANCE_ITEM_DIALOG_PADDING_BETWEEN_TEXT_FIELDS.dp))
+
+                MaintenanceTextField(
+                    fieldValue = location,
+                    fieldLabel = R.string.location_field_label,
+                    placeholder = R.string.maintenance_item_dialog_location_field_placeholder,
+                    focusAction = FocusOptions.Clear
+                )
+
+                Spacer(modifier = Modifier.padding(MAINTENANCE_ITEM_DIALOG_LOCATION_FIELD_BOTTOM_PADDING.dp))
+
+                Row(
+                    modifier = Modifier
+                        .fillMaxWidth(),
+                    verticalAlignment = Alignment.CenterVertically,
+                    horizontalArrangement = Arrangement.SpaceEvenly
+                ) {
+                    ExtendedFloatingActionButton(
+                        onClick = {
+                            isDialogOpen.value = false
```

Comment:
could break this out into an onMaintenanceDialogClosed function and call it from here and in the FAB below